src/dataExplorer/components/Results.tsx

```diff
@@ -25,31 +25,39 @@ import {FluxResult} from 'src/types/flows'
 
 import './Results.scss'
 import {isFlagEnabled} from 'src/shared/utils/featureFlag'
+import {bytesFormatter} from 'src/shared/copy/notifications'
 
 // simplified version migrated from src/flows/pipes/Table/view.tsx
 const QueryStat: FC = () => {
   const {result} = useContext(ResultsContext)
 
   const tableColumn = result?.parsed?.table?.getColumn('table') || []
-  const lastTableValue = tableColumn[tableColumn.length - 1] || -1
+  const lastTableValue = tableColumn[tableColumn.length - 1]
 
   let tableNum = 0
 
   if (typeof lastTableValue === 'string') {
     tableNum = parseInt(lastTableValue) + 1
   } else if (typeof lastTableValue === 'boolean') {
-    tableNum = lastTableValue ? 1 : 0
-  } else {
-    // number
+    console.error('Cannot extract tableId. Check parsed csv output.')
+  } else if (typeof lastTableValue === 'number') {
     tableNum = lastTableValue + 1
   }
 
   return (
-    <div className="query-stat">
-      <span className="query-stat--bold">{`${tableNum} tables`}</span>
-      <span className="query-stat--bold">{`${
-        result?.parsed?.table?.length || 0
-      } rows`}</span>
+    <div className="query-stat" data-testid="query-stat">
+      {result.truncated ? (
```

Comment:
```suggestion
      {result?.truncated ? (
```

`result` becomes undefined when the "+ New Script" button is clicked

Reply:
nice catch. Thank you!